pq-crypto/sike_r3/sikep434r3_sidh.c

```diff
@@ -0,0 +1,329 @@
+// -----------------------------------------------------------------------------
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Supersingular Isogeny Key Encapsulation Library
+//
+// Abstract: ephemeral supersingular isogeny Diffie-Hellman key exchange (SIDH)
+// -----------------------------------------------------------------------------
+
+#include "sikep434r3.h"
+#include "sikep434r3_fpx.h"
+#include "sikep434r3_ec_isogeny.h"
+#include "sikep434r3_api.h"
+#include "../../include/openssl/rand.h"
+
+
+// Initialization of basis points
+static void init_basis(const digit_t *gen, f2elm_t *XP, f2elm_t *XQ, f2elm_t *XR)
+{
+    fpcopy(gen,                  XP->e[0]);
+    fpcopy(gen +   SIKE_P434_R3_NWORDS_FIELD, XP->e[1]);
+    fpcopy(gen + 2*SIKE_P434_R3_NWORDS_FIELD, XQ->e[0]);
+    fpcopy(gen + 3*SIKE_P434_R3_NWORDS_FIELD, XQ->e[1]);
+    fpcopy(gen + 4*SIKE_P434_R3_NWORDS_FIELD, XR->e[0]);
+    fpcopy(gen + 5*SIKE_P434_R3_NWORDS_FIELD, XR->e[1]);
+}
+
+// Generation of Bob's secret key
+// Outputs random value in [0, 2^Floor(Log(2, oB)) - 1]
+int random_mod_order_B(unsigned char* random_digits)
+{
+    //POSIX_GUARD_RESULT(get_random_bytes(random_digits, SIKE_P434_R3_SECRETKEY_B_BYTES));
+    //POSIX_GUARD(RAND_bytes(random_digits, SIKE_P434_R3_SECRETKEY_B_BYTES));
+    if (RAND_bytes(random_digits, SIKE_P434_R3_SECRETKEY_B_BYTES) != 1) {
+        // FAILURE
+        return 0;
+    }
+    random_digits[SIKE_P434_R3_SECRETKEY_B_BYTES-1] &= SIKE_P434_R3_MASK_BOB; // Masking last byte
+
+    // SUCCESS
+    return 1;
+}
+
+// Alice's ephemeral public key generation
+// Input:  a private key PrivateKeyA in the range [0, 2^eA - 1].
+// Output: the public key PublicKeyA consisting of 3 elements in GF(p^2) which are encoded
+//         by removing leading 0 bytes.
+int EphemeralKeyGeneration_A(const unsigned char* PrivateKeyA, unsigned char* PublicKeyA)
+{
+    //point_proj_t R, phiP = {0}, phiQ = {0}, phiR = {0}, pts[SIKE_P434_R3_MAX_INT_POINTS_ALICE];
```

Comment:
If I remember correctly, this might have been used to address the brackets error that you were running into in Cmake? You might want to leave a comment here and for the instances you commented out the line, so we can keep track why this was done. Leaving a comment at the top of file explaining the reason for all the commented out initializations would suffice too.